flow/project.py

```diff
@@ -236,6 +242,76 @@ def make_bundles(operations, size=None):
     return _make_bundles(operations, size)
 
 
+class aggregate(object):
+    """Decorator for operation functions that are aggregate operations.
+
+    .. code-block:: python
+        example_aggregate = aggregate()
+
+        @example_aggregate
+        @FlowProject.operation
+        def foo(jobs):
+            return len(jobs)
+
+    :param groupper:
+        Information on how to aggregate jobs.
+    :type groupper:
+        callable
+    """
+
+    def __init__(self, grouper=None):
+        if grouper is None:
+            def grouper(jobs):
+                yield jobs
+
+        self.grouper = grouper
+
+    @classmethod
+    def groupsof(cls, num=1, fillvalue=None):
```

Comment:
How should the `fillvalue` attribute be treated by the users? What all values should it take? Should we check pre/post conditions for that particular value?

Reply:
I think it's best to skip these.
For instance, if I want a group of 3 and I get [joba, None, None] then it's was probably supposed to happen. Hence we shouldn't interfere much when a use asks for "user specified" arguments